compiler/aarch64/codegen/BinaryEvaluator.cpp

```diff
@@ -30,6 +30,113 @@
 #include "il/Node_inlines.hpp"
 #include "infra/Bit.hpp"
 
+/**
+ * Helper that evaluates a child node.
+ * @param[in] childReg : the register of the child node being looked at
+ * @param[in] trgReg : the target register
+ * @param[in] child : the child node being looked at
+ * @param[in] parent : the parent of the child node being looked at
+ * @param[in] cg : codegenerator
+ */
+static inline void
+evaluateRegister(TR::Register **childReg, TR::Register *trgReg, TR::Node *child, TR::Node *parent, TR::CodeGenerator *cg)
+   {
+   int64_t value;
+	if (child->getOpCode().isLoadConst() && child->getRegister() == NULL)
+      {
+      value = child->getLongInt();
+      //if(constantIsUnsignedImm12(value))
+         //{
+         //TODO: Cover special cases for immediate loads
+         //}
+      //else
+         //{
+         *childReg = cg->allocateRegister();
+         loadConstant64(cg, parent, value, *childReg);
+         //}
+      }
+   else
+      {
+      *childReg = cg->evaluate(child);
+      }
+   if(child->getReferenceCount() == 1 && NULL == trgReg)
+      {
+      trgReg = *childReg;
```

Comment:
It looks like the intent is to pass `trgReg` back to the caller.  If that's the case then I think the parameter should be declared as a `TR:Register **trgReg`, otherwise just the local pointer is updated.